### ISSUE TITLE
Exclude top-level directories of the backup source via relative path

### ISSUE THREAD

**Issue by sebschlicht** [restic/restic#3007]
Output of `restic version`
--------------------------
restic 0.8.3
compiled with go1.10 on linux/amd64

What should restic do differently? Which functionality do you think we should add?
----------------------------------------------------------------------------------
Paths in exclude files are applied to the full path of a file.
Currently there seems to be no solution to exclude only top-level directories in the backup source folder(s) without specifying their full paths.

It should be possible to exclude a top-level directory of the backup source, e.g. by a path  that's relative to the backup source folder such as `./Downloads`.

What are you trying to do? What problem would this solve?
---------------------------------------------------------
I want to backup multiple directories that have similar layouts, e.g.

    backup-source1/
    -- Downloads/
    -- Pictures/
    -- .cache/
    backup-source2/
    -- Downloads/
    ...

These source directories could be the home directories of the system users, for example.

While I want to backup all of them, let's assume I'd like to exclude all the `Downloads` folders.
Currently I can only exclude the folder on any level via `Downloads` or exclude each folder by specifying the absolute paths, individually. The first option is semantically wrong and the second contains redundancy and is harder to maintain, with quadratic growth (per source and excluded folder) of effort.

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
I'm sick of the backup topic and that the good tools either are incremental or has your current files navigatable - but not both.
Your project could not only replace my custom solution and finally provide a slick interface to this complex topic, which almost feels like working with `git`, but also offers both incremental backups as well as the ability to mount the current snapshot and have it browsable. Neat!

**Comment by MichaelEischer** [comment]
@sebschlicht Your restic version is quite old, the current version is 0.10.0.

That said excludes relative to the backup source folders still don't exit AFAIK. For your home folder example an exclude of `/home/*/Downloads` would also work. If there are home folders at a different location, that obviously doesn't help though.

**Comment by sebschlicht** [comment]
@MichaelEischer Yeah, I wasn't too sure whether this feature wasn't added in the meantime but my research suggested it didn't.

Thanks for the workaround, I didn't fully internalize that `*` isn't matched against the path separator yet!

Still I feel like having the ability to point at the backup source root would be beneficial. It allows to move the source root without changing the exclude file and to perform backups without knowing the absolute paths at all.

**Comment by Devocub** [comment]
Spent 20 minutes testing why folder still included turns out you need to specify full filesystem path
Incredible how this basic feature is not implemented

**Comment by bitnom** [comment]
wow I thought I had been excluding but then realized it's not. Is there some big difficulty supporting glob? Now I have to make a script that creates a massive exclude file

**Comment by rmst** [comment]
The docs (https://restic.readthedocs.io/en/latest/040_backup.html#excluding-files) are not clear about this behaviour. The behaviour for paths starting with / is made clear but not so for paths starting without it. Having relative-to-backup-root paths would also have been nice for backups that are made from APFS/BTRFS/ZFS snapshots (which seems like the proper way to do backups). In this case using absolute paths is impractical because the mount dir of the temporary snapshot is probably not stable. One workaround for this is to use a custom environment variable like ROOT and prefix all the ignore paths with `$ROOT/`.

**Comment by leaumar** [comment]
Came here because I was left with the same question after reading the docs: are relative paths in excludes/includes interpreted from pwd or from each backup path? So I guess the answer is from pwd, i.e. excluding ./Downloads will exclude it from the dir you're running in, not from the paths you're backing up.

I have some other gripes with the docs, like the statement "Patterns [...] are tested against the full path of a file/dir to be saved" right next to filters like `*.tmp` being shown to effectively behave like `**/*.tmp`. I now get that the statement is about relative paths being resolved to absolute ones before filters are checked, not that a filter must match a path from start to end. It was tripping me up for a while, every tool gives slightly different spins to partial or relative paths and leading or trailing slashes, so it's hard to trust any filter or doc without a lot of tricky testing.
